modules/mono/editor/GodotTools/GodotTools/Build/MsBuildFinder.cs

```diff
@@ -161,8 +161,21 @@ private static string FindMsBuildToolsPathOnWindows()
 
             // Try to find 15.0 with vswhere
 
-            string vsWherePath = Environment.GetEnvironmentVariable(Internal.GodotIs32Bits() ? "ProgramFiles" : "ProgramFiles(x86)");
-            vsWherePath += "\\Microsoft Visual Studio\\Installer\\vswhere.exe";
+            var envNames = Internal.GodotIs32Bits() ? new[] { "ProgramFiles", "ProgramW6432" } : new[] { "ProgramFiles(x86)", "ProgramFiles" };
```

Comment:
Wouldn't it make sense to check the matching architecture's path first? This would mean that the `"ProgramFiles"` environment variable is checked first, since for a 32-bit process it points to 32-bit Program Files, and for a 64-bit process it points to 64-bit Program Files.

```suggestion
            var envNames = Internal.GodotIs32Bits() ? new[] { "ProgramFiles", "ProgramW6432" } : new[] { "ProgramFiles", "ProgramFiles(x86)" };
```

We might be able to optimize this further by only changing the fallback environment variable. I haven't compiled this but I wonder if this works:

```suggestion
            var envNames = new[] { "ProgramFiles", Internal.GodotIs32Bits() ? "ProgramW6432" : "ProgramFiles(x86)" };
```

Reply:
What I have heard is, in windows 64 bit, Visual Studio Installer most likely be installed in `ProgramFiles(x86)` by default. Only me or some people that change the default path of installer would met this problem. So I think it would be more efficient to ordering this way

Reply:
Yes, I'm just thinking of future proofing it, in case there is a 64-bit Visual Studio Installer in the future and then it would be placed in the 64-bit folder instead (and if both exist, prefer the 64-bit one). I'll let @neikeq decide what's best here.